lib/hyperswitch.js

```diff
@@ -52,8 +51,9 @@ function HyperSwitch(options, req, parOptions) {
         this._priv = par._priv;
         this.config = this._priv.options.conf;
         this._rootReq = par._rootReq || req;
-        this._forwardedHeaders = par._forwardedHeaders || this._rootReq.headers;
-        this._authService = par._authService ? new AuthService(par._authService) : null;
+        this._requestFilters = par._requestFilters;
+        this._subRequestFilters = par._subRequestFilters;
+        this.ctx = Object.assign({}, par.ctx);
```

Comment:
Hmmm.. if we want full isolation, we should probably make a deep copy here. That might be too expensive, so I wonder if cloning is actually worth it here. Do you have use cases in mind that need the isolation?